src/multirust-dist/src/dist.rs

```diff
@@ -21,26 +21,26 @@ pub const UPDATE_HASH_LEN: usize = 20;
 
 #[derive(Debug)]
 pub struct ToolchainDesc {
-    pub arch: Option<String>,
-    pub os: Option<String>,
-    pub env: Option<String>,
     // Either "nightly", "stable", "beta", or an explicit version number
```

Comment:
Right now the regex is very simple. It accepts these two forms: `1.9.0` or `0.10.0`, with exactly those numbers of digits. The next stable release will publish channel manifests for the version number, and it should just work to install `rustup update 1.8.0`, add/remove targets from it and everything.